### ISSUE TITLE
http client fails with ring/ring-jetty-adapter 1.11.0

### ISSUE THREAD

**Issue by dharrigan** [cognitect-labs/aws-api#250]
Thank you for your interest in helping to improve Cognitect's aws-api!

## Dependencies

`Clojure CLI version 1.11.1.1435`

```
openjdk 21.0.1 2023-10-17 LTS
OpenJDK Runtime Environment Temurin-21.0.1+12 (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.1+12 (build 21.0.1+12-LTS, mixed mode, sharing)
```

```clojure
ring/ring-jetty-adapter {:mvn/version "1.11.0"}
com.cognitect.aws/api {:mvn/version "0.8.686"}
com.cognitect.aws/cloudfront {:mvn/version "847.2.1365.0"}
com.cognitect.aws/endpoints {:mvn/version "1.1.12.626"}
com.cognitect.aws/s3 {:mvn/version "848.2.1413.0"}
```

## Description with failing test case

As ring is now available as a `1.11.0` [release](https://github.com/ring-clojure/ring/blob/master/CHANGELOG.md) (along with the dependency `ring/ring-jetty-apapter 1.11.0`), when upgrading deps and then starting the http-client (as a dependency injected resource), the following occurs:

```
user=> (dev/go)
Syntax error (NoSuchFieldException) compiling . at (cognitect/http_client.clj:41:9).
getProperties
user=> 
```

Reverting back to `ring/ring-jetty-adapter {:mvn/version "1.10.0"}` fixes the problem.

Thank you.

-=david=-

**Comment by ovistoica** [comment]
Any idea when the transitive dependency will be solved? Currently we cannot use `aws-api` as we transitioned to https://github.com/sunng87/ring-jetty9-adapter which uses Jetty 12.0.x and we cannot downgrade or switch back to ring adapter as we rely on functionality from Jetty 12.0

I saw there was an effort to move the HTTP client to the JDK version

As an aside, Is there any example code of how to use the `:http-client` prop. I see it needs to implement the `HttpClient` protocol

```clojure
(defprotocol HttpClient
  (-submit [_ request channel]
    "Submit an http request, channel will be filled with response. Returns ch.

     Request map:

     :scheme                 :http or :https
     :server-name            string
     :server-port            integer
     :uri                    string
     :query-string           string, optional
     :request-method         :get/:post/:put/:head/:delete
     :headers                map from downcased string to string
     :body                   ByteBuffer, optional
     :timeout-msec           opt, total request send/receive timeout
     :meta                   opt, data to be added to the response map

     content-type must be specified in the headers map
     content-length is derived from the ByteBuffer passed to body

     Response map:

     :status            integer HTTP status code
     :body              ByteBuffer, optional
     :headers           map from downcased string to string
     :meta              opt, data from the request

     On error, response map is per cognitect.anomalies.

     Alpha. This will absolutely change.")
  (-stop [_] "Stops the client, releasing resources"))
```

**Comment by tobias** [comment]
I have a fork of `com.cognitect/http-client` that uses Jetty 11: https://github.com/tobias/cognitect-http-client.  I'm currently using it in production for clojars.org without issue. Folks are free to use that, and @scottbale you are welcome to use the changes there if you do decide to upgrade the official client to Jetty 11.

**Comment by bhurlow** [comment]
+1 on this, we can't use `aws-api` with https://github.com/hyperfiddle/electric since it requires Jetty 11 for websocket support

**Comment by tobias** [comment]
@bhurlow Does my fork work for you?

**Comment by reify-kurt-wheeler** [comment]
> Solving the problem of aws-api's transitive dependency on jetty 9.4.x is our top priority for our next release.

Hello, it's been about 7 months since this comment. Is there any solution for this issue yet?

**Comment by scottbale** [comment]
We have just [released a beta release](https://github.com/cognitect-labs/aws-api/blob/main/CHANGES.md#08710-beta01--2024-10-16) which introduces a Java-native http client. With this release, Jetty is no longer by default a transitive dependency.

@dharrigan if time permits, could you please verify whether this release solves your issue? Thanks.

(I will leave this issue open for the time being until we verify.)

**Comment by scottbale** [comment]
Fixed in version 0.8.711
